### PR TITLE
Feature: exclude small mask

### PR DESCRIPTION
**Pull request by v-imark**

### CODE REVIEW

File: segment_images.py [hunk 1]
```diff
@@ -44,12 +47,17 @@
...
     default="test",
 )
-
 parser.add_argument(
     "--output",
     help="Path to where the output will be stored",
     default="./outputs",
 )
+parser.add_argument(
+    "--thresh",
```
Comment (ResearchDaniel) [inline segment_images.py]:
```suggestion
    "--min_area",
```

File: segment_images.py [hunk 2]
```diff
@@ -100,7 +112,9 @@ def main(args):
 
     img_folder = Path(args.data_path, args.dataset, args.split)
 
-    output_name = f"{args.dataset}_{args.split}_{args.size}"
+    output_name = (
+        f"{args.dataset}_{args.split}_{args.size}_{args.points_per_side}_{args.thresh}"
```
Comment (ResearchDaniel) [inline segment_images.py]: Might want to add parameter names
```suggestion
        f"{args.dataset}_{args.split}_{args.size}_points-per-side_{args.points_per_side}_min-area_{args.thresh}"
```

File: utils/exclude_small_masks.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+def exclude_small_masks(masks, threshold=0.01):
+    """Excludes masks whose size is less than threshold*size of corresponding image."""
+    filtered_masks = []
+    min_area = masks[0]["crop_box"][2] * masks[0]["crop_box"][3] * threshold
+    for mask in masks:
+        if mask["area"] >= min_area:
+            filtered_masks.append(mask)
+
+    return sorted(filtered_masks, key=lambda x: x["area"], reverse=True)
```
Comment (ResearchDaniel) [inline utils/exclude_small_masks.py]: Why sort them?

File: utils/save_filter_data.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import csv
+from pathlib import Path
+
+
+def save_filter_data(total, filtered, thresh, output_path):
```
Comment (ResearchDaniel) [inline utils/save_filter_data.py]: This is more about saving settings and aggregated data?
```suggestion
def save_segmentation_info(total, filtered, thresh, output_path):
```

Comment (ResearchDaniel) [reply utils/save_filter_data.py]: Not sure if it makes too much sense to have a csv file with only one entry.
Store info about all image segmentations in a single json/csv file.

Depends on what you want to do with the info later.

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

File: utils/save_filter_data.py (hunk 4 above)
Comment (v-imark) [reply utils/save_filter_data.py]: Changed it now so all info is tored in a single json file. The thought was to see how useful the excluding etc. is.

**Review by ResearchDaniel — APPROVED** [review]

**Comment by ResearchDaniel** [comment]
Note: During the analysis it will probably make sense to use `min-area=0` to allow for understanding which parameter that makes sense for a given data set

**Pull request merged**